geaflow/geaflow-dsl/geaflow-dsl-connector/geaflow-dsl-connector-jdbc/src/main/java/com/antgroup/geaflow/dsl/connector/jdbc/JDBCTableSource.java

```diff
@@ -0,0 +1,182 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.antgroup.geaflow.dsl.connector.jdbc;
+
+import com.antgroup.geaflow.api.context.RuntimeContext;
+import com.antgroup.geaflow.common.config.Configuration;
+import com.antgroup.geaflow.dsl.common.data.Row;
+import com.antgroup.geaflow.dsl.common.exception.GeaFlowDSLException;
+import com.antgroup.geaflow.dsl.common.types.StructType;
+import com.antgroup.geaflow.dsl.common.types.TableSchema;
+import com.antgroup.geaflow.dsl.connector.api.FetchData;
+import com.antgroup.geaflow.dsl.connector.api.Offset;
+import com.antgroup.geaflow.dsl.connector.api.Partition;
+import com.antgroup.geaflow.dsl.connector.api.TableSource;
+import com.antgroup.geaflow.dsl.connector.api.Windows;
+import com.antgroup.geaflow.dsl.connector.api.serde.TableDeserializer;
+import com.antgroup.geaflow.dsl.connector.api.serde.impl.RowTableDeserializer;
+import com.antgroup.geaflow.dsl.connector.jdbc.util.JDBCUtils;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JDBCTableSource implements TableSource {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JDBCTableSource.class);
+    private Configuration tableConf;
+    private StructType schema;
+    private String driver;
+    private String url;
+    private String username;
+    private String password;
+    private String tableName;
+    private Connection connection;
+    private Statement statement;
+
+    @Override
+    public void init(Configuration tableConf, TableSchema tableSchema) {
+        LOGGER.info("prepare with config: {}, \n schema: {}", tableConf, tableSchema);
+        this.tableConf = tableConf;
+        this.schema = tableSchema;
+
+        this.driver = tableConf.getString(JDBCConfigKeys.GEAFLOW_DSL_JDBC_DRIVER);
+        this.url = tableConf.getString(JDBCConfigKeys.GEAFLOW_DSL_JDBC_URL);
+        this.username = tableConf.getString(JDBCConfigKeys.GEAFLOW_DSL_JDBC_USERNAME);
+        this.password = tableConf.getString(JDBCConfigKeys.GEAFLOW_DSL_JDBC_PASSWORD);
+        this.tableName = tableConf.getString(JDBCConfigKeys.GEAFLOW_DSL_JDBC_TABLE_NAME);
+    }
+
+    @Override
+    public void open(RuntimeContext context) {
+        try {
+            Class.forName(this.driver);
+            this.connection = DriverManager.getConnection(url, username, password);
+            this.statement = connection.createStatement();
+        } catch (Exception e) {
+            throw new GeaFlowDSLException("failed to connect to database", e);
+        }
+    }
+
+    @Override
+    public List<Partition> listPartitions() {
+        return Collections.singletonList(new JDBCPartition(this.tableName));
```

Comment:
Can we set the partitions by a config like "geaflow.dsl.jdbc.partition.num". So that we can read the jdbc concurrently. User can specified a field to split the table. e.g. user specify the "id" as the split field, then each partition can read the jdbc by the sql "select * from t where id >= {partition_start} and id < {partition_end}. 
you can refer the jdbc datasource in spark: https://spark.apache.org/docs/latest/sql-data-sources-jdbc.html

Reply:
I tried to add the configKey `geaflow.dsl.jdbc.partition.num` and wrote an SQL query that uses this configKey, but I encountered an error: `org.apache.calcite.sql.parser.impl.ParseException: Encountered ". partition" at line 10, column 48.` When I change `partition` to other words (such as `partitio`), the query runs fine. Is it because `partition` is a special word in SQL, or is this a bug?

Reply:
> I tried to add the configKey `geaflow.dsl.jdbc.partition.num` and wrote a SQL query that uses this configKey, but I encountered an error: `org.apache.calcite.sql.parser.impl.ParseException: Encountered ". partition" at line 10, column 48.`. However, there is no space between `.` and `partition` in the SQL. When I change `partition` to other words (such as `partitio`), the query runs fine. Is it because `partition` is a special word in SQL, or is this a bug?

PARTITION is a reserved keyword in sql, use <geaflow.dsl.jdbc.`partition`.num> instead